lib/stupidedi/transaction_sets/004010/implementations/IN810.rb

```diff
@@ -0,0 +1,191 @@
+                                                                       b::Element(e::Situational, "Address Information"),
+                                                                       b::Element(e::Situational, "Address Information")),
+                                                            b::Segment(140, s:: N4, "Geographic Location", r::Situational, d::RepeatCount.unbounded,
+                                                                       b::Element(e::Situational, "City Name"),
+                                                                       b::Element(e::Situational, "State or Province Code"),
+                                                                       b::Element(e::Situational, "Postal Code"),
+                                                                       b::Element(e::Situational, "Country Code"),
+                                                                       b::Element(e::Situational, "Location Qualifier"),
+                                                                       b::Element(e::Situational, "Location Identifier"))),
+                                           b::Segment(200, s:: ITD, "Terms of Sale or Deferred Terms of Sale", r::Situational, d::RepeatCount.unbounded,
+                                                      b::Element(e::Situational, "Terms Type Code"),
+                                                      b::Element(e::Situational, "Terms Basis Date Code"),
+                                                      b::Element(e::Situational, "Terms Discount Percent"),
+                                                      b::Element(e::Situational, "Terms Discount Due Date"),
+                                                      b::Element(e::Situational, "Terms Discount Days Due"),
+                                                      b::Element(e::Situational, "Terms Net Due Date"),
+                                                      b::Element(e::Situational, "Terms Net Days"),
+                                                      b::Element(e::Situational, "Terms Discount Amount"),
+                                                      b::Element(e::Situational, "Terms Deferred Due Date"),
+                                                      b::Element(e::Situational, "Deferred Amount Due"),
+                                                      b::Element(e::Situational, "Percent of Invoice Payable"),
+                                                      b::Element(e::Situational, "Description"),
+                                                      b::Element(e::Situational, "Day of Month"),
+                                                      b::Element(e::Situational, "Payment Method Code"),
+                                                      b::Element(e::Situational, "Percent"))
+                        ), #end TableDef HEADING
```

Comment:
Closing parentheses should not be on their own line. You can leave the `# end XYZ` markers. In cases where there are several consecutive closing parentheses, they need to be written side-by-side; multiple end markers would need to be joined like so: `...))) # end A, B, C`